bindings/python/fdb/tuple.py

```diff
@@ -306,6 +307,14 @@ def _reduce_children(child_values):
     return bytes_list, version_pos
 
 
+if sys.version_info < (2, 7):
+    def _bit_length(x):
+        return len(bin(x)) - 2
```

Comment:
This returns different values than `bit_length()` for 0 and negative numbers. I don't think we use this on 0, but we do use it on negative numbers. We could fix this by using this implementation, described in the [documentation](https://docs.python.org/3/library/stdtypes.html#int.bit_length) of `bit_length()` as being equivalent:

```
def _bit_length(x):
    s = bin(x)       # binary representation:  bin(-37) --> '-0b100101'
    s = s.lstrip('-0b') # remove leading zeros and minus sign
    return len(s)
```

Or maybe we could do something like this, if we tested it and found it to be faster:

```
def _bit_length(x):
    if x == 0:
        return 0 
    return len(bin(abs(x)))-2
```

Reply:
Oof, nice catch. Done. (I also opened https://github.com/amzn/ion-python/issues/69, so they can benefit too)